protocol/messenger_activity_center.go

```diff
@@ -318,11 +399,126 @@ func (m *Messenger) handleActivityCenterAccepted(state *ReceivedMessageState, me
 }
 
 func (m *Messenger) handleActivityCenterDismissed(state *ReceivedMessageState, message protobuf.SyncActivityCenterDismissed) error {
-	resp, err := m.DismissActivityCenterNotifications(context.TODO(), toHexBytes(message.Ids), false)
+	resp, err := m.DismissActivityCenterNotifications(context.TODO(), toHexBytes(message.Ids), message.Clock, false)
 
 	if err != nil {
 		return err
 	}
 
 	return state.Response.Merge(resp)
 }
+
+func (m *Messenger) handleSyncActivityCenterNotificationState(state *ReceivedMessageState, a *protobuf.SyncActivityCenterNotificationState) error {
+	s := &ActivityCenterState{
+		HasSeen:   a.HasSeen,
+		UpdatedAt: a.UpdatedAt,
+	}
+	_, err := m.persistence.UpdateActivityCenterNotificationState(s)
+	if err != nil {
+		return err
+	}
+	state.Response.SetActivityCenterState(s)
```

Comment:
@qfrank we will get to this line even `s.UpdatedAt` is equal or less then the `updated_at` column at the db, which should not be the case, cause sending this signal will cause UI updated on the client side. It should be sent only when the db is really updated.